python/ray/rllib/models/preprocessors.py

```diff
@@ -193,7 +193,7 @@ def _init_shape(self, obs_space, options):
 
     @override(Preprocessor)
     def transform(self, observation):
-        self.check_shape(observation)
+        self.check_shape(tuple(observation))
```

Comment:
Could you move this to inside the check function here? https://github.com/ray-project/ray/blob/master/python/ray/rllib/models/preprocessors.py#L54

The reason is that the tuple conversion could be expensive, so we should only do it if inside the validation interval if block.